Include/moduleobject.h

```diff
@@ -48,11 +48,13 @@ typedef struct PyModuleDef_Base {
   PyObject* m_copy;
 } PyModuleDef_Base;
 
-#define PyModuleDef_HEAD_INIT { \
-    PyObject_HEAD_INIT(NULL)    \
-    NULL, /* m_init */          \
-    0,    /* m_index */         \
-    NULL, /* m_copy */          \
+// TODO(eduardo-elizondo): This is only used to simplify the review of GH-19474
+// Rather than changing this API, we'll introduce PyModuleDef_HEAD_IMMORTAL_INIT
+#define PyModuleDef_HEAD_INIT {       \
```

Comment:
Please don't change module objects. This may cause huge memory leak when user reload modules many times in interactive shell.

Reply:
Do you changing many objects only for measuring performance?
I added "DO-NOT-MERGE" label to block this PR is accidentally merged.

Reply:
> Please don't change module objects. This may cause huge memory leak when user reload modules many times in interactive shell.

Do you have an example where this will be a leak? This is not changing module objects but the module definition struct. It should not have any impact in the impact of any imported module regardless of which import machinery is used.

To showcase this, I locally exposed in module object's member defs:
`{"__my_spec__", T_OBJECT, offsetof(PyModuleObject, md_def), READONLY},`
Then:
```
>>> import sys; import gc
>>> sys.getrefcount(gc)
21
>>> sys.getrefcount(gc.__my_spec__)
1152921504606846976
>>> import gc; import gc; import gc
>>> sys.getrefcount(gc)
21
>>> sys.getrefcount(gc.__my_spec__)
1152921504606846976
```

Reply:
> Do you changing many objects only for measuring performance?

Not really at this point. Just using this to mark objects that we explicitly know that are immortals like the static types and the singletons

I can revert this change if you'd like though

Reply:
>  This is not changing module objects but the module definition struct. It should not have any impact in the ref count of any imported module regardless of which import machinery is used.

Sorry, I was wrong.